src/main/java/org/blockchain_innovation/factom/identiy/did/json/CustomServiceSerializer.java

```diff
@@ -0,0 +1,27 @@
+package org.blockchain_innovation.factom.identiy.did.json;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import org.factomprotocol.identity.did.model.Service;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CustomServiceSerializer implements JsonSerializer<Service> {
+
+    @Override
+    public JsonElement serialize(Service service, Type type, JsonSerializationContext jsonSerializationContext) {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.add("id", jsonSerializationContext.serialize(service.getId()));
+        jsonObject.add("type", jsonSerializationContext.serialize(service.getType()));
+        jsonObject.add("serviceEndpoint", jsonSerializationContext.serialize(service.getServiceEndpoint()));
+        jsonObject.add("priorityRequirement", jsonSerializationContext.serialize(service.getPriorityRequirement()));
+        service.keySet().forEach(key -> {
```

Comment:
Should not be possible in reality. But have you checked what happens if I shadow for instance one of the properties in the hashmap? We might want to make an explicit check for that

Reply:
Currently this:
```
Service service = new Service()
                .id("something");
service.put("id", "something-else");
```
Returns a JSON of:
```
{
  "id": "something-else"
}```
So the previous value is just overridden if we try to add a new key to the map. To me this made sense, as the `service.setId` method and `service.put("id", ...)` essentially have the same functionality. Should we be throwing an error here instead?

Reply:
Yes. The map is for freeform (additional data) and not subject to any validation or the contract in general. I would certainly add protection that people do not pass in data that really should be in the properties, because these values are what the rest of the code looks for.

Reply:
Updated.